### PR TITLE
feat: `gno test` support external pkgs

### PR DESCRIPTION
**Pull request by harry-hov**
Also look for pkg/realm in `GNOHOME` (pkgs fetched from chain via `mod download`)

Now developers can run `gno mod download` before `gno test` to support packages that are not part of `/examples` directory

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,8 @@ install_gnokey:
 install_gnodev:
 	@echo "Installing gnodev"
 	go install ./cmd/gnodev
+	@echo "Copying stdlibs"
+	rsync -ur stdlibs $${HOME}/.gno
```
Comment (moul) [inline Makefile:50]: What about embedding `stdlibs` in binary with `go:embed`?

File: cmd/gnodev/test.go [hunk 2]
```diff
@@ -114,7 +123,7 @@ func execTest(cfg *testCfg, args []string, io *commands.IO) error {
 
 	// guess opts.RootDir
 	if cfg.rootDir == "" {
-		cfg.rootDir = guessRootDir()
```
Comment (moul) [inline cmd/gnodev/test.go:117]: I like this :)

File: cmd/gnodev/util.go [hunk 3]
```diff
@@ -114,6 +115,28 @@ func guessRootDir() string {
...
+	for {
+		filePath := filepath.Join(absPath, "gno.mod")
+		_, err := os.Stat(filePath)
+		if err == nil {
+			return filePath, nil
+		}
+
+		absPath = filepath.Dir(absPath)
+		// Stop iterating if we have reached the root directory
+		if absPath == "/" {
```
Comment (moul) [inline cmd/gnodev/util.go:132]:
```suggestion
		if absPath == previousAbsPath {
```

For windows or other strange cases.

**Comment by moul** [comment]
TLDR is:
* we want to remove `--root-dir` (git clone guessing)
* downloading `stdlibs` VS embedding `stdlibs`?

**Comment by moul** [comment]
@harry-hov can you checkout https://github.com/gnolang/gno/pull/1014, I think you should collaborate to make your work merged.

**Comment by gfanton** [comment]
It may be a small detail, but based on the `go env` command, might it make more sense to call these two variables **GNOPATH** (for containing dependencies) and **GNOROOT** (for containing `gno`, `tool`, and `stdlib`) to achieve better symmetry with Go's naming conventions? While the **GOPATH** name might seem less relevant with the adoption of Go modules, it's still the place for storing Go dependencies. I may be wrong, but I believe it would feel more natural for newcomers to encounter **GNOPATH** and **GNOROOT** instead of **GNO_HOME** and **GNO_ROOT**. Even though **GNO_HOME** does make sense, the proposed names would align more closely with established patterns in the Go community.

**Comment by thehowl** [comment]
Closing as draft for 1yr+. Also, I think it's misguided to begin with.
